scenario/context/CometContext.ts

```diff
@@ -248,12 +260,58 @@ export class CometContext {
     await this.world.hre.network.provider.send('hardhat_mine', [`0x${blocks.toString(16)}`]);
   }
 
+  async executePendingProposal(proposalId: number, startBlock: number, endBlock: number) {
+    const governor = await this.getGovernor();
+
+    const blockNow = await this.world.hre.ethers.provider.getBlockNumber();
+    const blocksUntilStart = startBlock - blockNow;
+    const blocksFromStartToEnd = endBlock - Math.max(startBlock, blockNow);
+
+    if (blocksUntilStart > 0) {
+      await this.mineBlocks(blocksUntilStart);
+    }
+
+    for (const voter of this.voters) {
+      try {
+        // Voting can fail if voter has already voted
+        const voterSigner = await this.world.impersonateAddress(voter);
+        const govAsVoter = await this.world.hre.ethers.getContractAt(
+          'IGovernorBravo',
+          governor.address,
+          voterSigner
+        ) as IGovernorBravo;
+        await this.setNextBaseFeeToZero();
+        await govAsVoter.castVote(proposalId, 1, { gasPrice: 0 });
+      } catch (err) {
+        debug(`Error while voting ${err}`);
+      }
+    }
+    await this.mineBlocks(blocksFromStartToEnd);
+
+    const adminSigner = await this.world.impersonateAddress(this.voters[0]);
```

Comment:
Just FYI, on my branch I added `value` arg to `impersonateAddress`, which for signers which *can* receive eth is slightly better than having to set fees/price to zero -- only really necessary if the call is in a function we would call as part of `deploy` and this works totally well for scenario-only code

Reply:
Noted, once merged, I can change that